src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3249 @@
+            "MSFT_CimConfigurationProviderRegistration",
+            "MSFT_PSConfigurationProviderRegistration",
+        };
+
+        // Create a HashSet for fast lookup. According to MSDN, the time complexity of search for an element in a HashSet is O(1)
+        private static readonly HashSet<string> s_hiddenResourceCache = new HashSet<string>(s_hiddenResourceList,
+            StringComparer.OrdinalIgnoreCase);
+
+        // a collection to hold current importing script based resource file
+        // this prevent circular importing case when the script resource existing in the same module with resources it import-dscresource
+        private static readonly HashSet<string> s_currentImportingScriptFiles = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// DSC class cache for this runspace.
+        /// Cache stores the DSCRunAsBehavior, cim class and boolean to indicate if an Inbox resource has been implicitly imported.
+        /// </summary>
+        private static Dictionary<string, DscClassCacheEntry> ClassCache
+        {
+            get
+            {
+                if (t_classCache == null)
+                {
+                    t_classCache = new Dictionary<string, DscClassCacheEntry>(StringComparer.OrdinalIgnoreCase);
+                }
+
+                return t_classCache;
+            }
+        }
+
+        [ThreadStatic]
+        private static Dictionary<string, DscClassCacheEntry> t_classCache;
+
+        /// <summary>
+        /// DSC classname to source module mapper.
+        /// </summary>
+        private static Dictionary<string, Tuple<string, Version>> ByClassModuleCache
+            => t_byClassModuleCache ??= new Dictionary<string, Tuple<string, Version>>(StringComparer.OrdinalIgnoreCase);
+
+        [ThreadStatic]
+        private static Dictionary<string, Tuple<string, Version>> t_byClassModuleCache;
+
+        /// <summary>
+        /// DSC filename to defined class mapper.
+        /// </summary>
+        private static Dictionary<string, IEnumerable<PSObject>> ByFileClassCache
+            => t_byFileClassCache ??= new Dictionary<string, IEnumerable<PSObject>>(StringComparer.OrdinalIgnoreCase);
+
+        [ThreadStatic]
+        private static Dictionary<string, IEnumerable<PSObject>> t_byFileClassCache;
+
+        /// <summary>
+        /// Filenames from which we have imported script dynamic keywords.
+        /// </summary>
+        private static HashSet<string> ScriptKeywordFileCache
+            => t_scriptKeywordFileCache ??= new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+        [ThreadStatic]
+        private static HashSet<string> t_scriptKeywordFileCache;
+
+        /// <summary>
+        /// Default ModuleName and ModuleVersion to use.
+        /// </summary>
+        private static readonly Tuple<string, Version> s_defaultModuleInfoForResource = new Tuple<string, Version>("PSDesiredStateConfiguration", new Version(1, 1));
```

Comment:
Isn't this should be 2.5 or 3.0?

Reply:
This is like stub module info for classes like base registrations that don't actually belong to a PS module.
It don't think anything actually relies on these values ; and considering that they were not updated as "PSDesiredStateConfiguration" module versions were going up, I'd like to keep this as is as long as this doesn't break anything.